### PR TITLE
partners: add mixedbread-ai integration

### PR DESCRIPTION
**Pull request by huangrpablo**
- [x] **PR message**: ***Delete this entire checklist*** and replace with
    - **Description:** Add support for mixedbread-ai's embeddding & reranking API.
    - **Dependencies:** mixedbread-ai="^2.2.2"
    - **Twitter handle:** mixedbreadai

- [x] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

### CODE REVIEW

**Comment by huangrpablo** [comment]
I will create a new branch to cleanup embaas

File: libs/partners/mixedbreadai/langchain_mixedbreadai/embeddings.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    truncation_strategy: TruncationStrategy = Field(default=TruncationStrategy.START)
+    normalized: bool = Field(default=True)
+    dimensions: Optional[int] = Field(default=None)
+    """"The desired number of dimensions in the output vectors. 
+    Only applicable for Matryoshka-based models"""
+    prompt: Optional[str] = Field(default=None)
+
+    def embed(
+        self,
+        texts: List[str],
```
Comment (juliuslipp) [inline libs/partners/mixedbreadai/langchain_mixedbreadai/embeddings.py:55]: Is this the interface? Would assume that this should only accept a single text.

File: libs/partners/mixedbreadai/langchain_mixedbreadai/embeddings.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                truncation_strategy=self.truncation_strategy,
+                normalized=self.normalized,
+                dimensions=self.dimensions,
+                prompt=self.prompt,
+                request_options=self._request_options,
+            )
+        ).data
+        return [item.embedding for item in embeddings]
+
+    def embed_documents(self, texts: List[str]) -> List[List[float]]:
```
Comment (juliuslipp) [inline libs/partners/mixedbreadai/langchain_mixedbreadai/embeddings.py:87]: Would assume this should accept documents

File: libs/partners/mixedbreadai/langchain_mixedbreadai/rerank.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            model=self.model,
+            query=query,
+            input=docs,
+            rank_fields=rank_fields,
+            top_k=self.top_n,
+            return_input=False,
+            request_options=self._request_options,
+        ).data
+
+        return [{"index": result.index, "score": result.score} for result in results]
```
Comment (juliuslipp) [inline libs/partners/mixedbreadai/langchain_mixedbreadai/rerank.py]: U could probably do result.dict(), but doesn't matter. So whatever you prefer

Comment (huangrpablo) [reply libs/partners/mixedbreadai/langchain_mixedbreadai/rerank.py]: fixed

**Pull request closed without merging**